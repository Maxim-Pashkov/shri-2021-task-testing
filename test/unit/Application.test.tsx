import React from 'react';
import { it, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import { getApplication } from './getApplication';
import { Application } from '../../src/client/Application';

describe('Application test', () => {
    it('navbar is exists', () => {
        const { application } = getApplication(() => <Application />, {});

        const { container } = render(application);

        expect(container.querySelector('.navbar')).toMatchSnapshot();
    });

    it('Cart link output count items in cart', () => {
        const cartState = {
            5: {
                name: '123',
                price: 10,
                count: 3,
            },
            10: {
                name: '1234',
                price: 15,
                count: 5,
            },
        };

        const { application } = getApplication(
            () => <Application />,
            cartState
        );

        const { container } = render(application);

        expect(container.querySelector('.Application-Menu')).toMatchSnapshot();
    });
});
