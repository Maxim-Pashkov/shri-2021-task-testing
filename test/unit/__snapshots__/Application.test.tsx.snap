// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application test Cart link output count items in cart 1`] = `
<div
  class="Application-Menu collapse navbar-collapse"
>
  <div
    class="navbar-nav"
  >
    <a
      class="nav-link"
      href="/catalog"
    >
      Catalog
    </a>
    <a
      class="nav-link"
      href="/delivery"
    >
      Delivery
    </a>
    <a
      class="nav-link"
      href="/contacts"
    >
      Contacts
    </a>
    <a
      class="nav-link"
      href="/cart"
    >
      Cart (2)
    </a>
  </div>
</div>
`;

exports[`Application test navbar is exists 1`] = `
<nav
  class="navbar navbar-expand-sm navbar-light bg-light"
>
  <div
    class="container"
  >
    <a
      class="Application-Brand navbar-brand"
      href="/"
    >
      Example store
    </a>
    <button
      aria-label="Toggle navigation"
      class="Application-Toggler navbar-toggler"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="Application-Menu collapse navbar-collapse"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-link"
          href="/catalog"
        >
          Catalog
        </a>
        <a
          class="nav-link"
          href="/delivery"
        >
          Delivery
        </a>
        <a
          class="nav-link"
          href="/contacts"
        >
          Contacts
        </a>
        <a
          class="nav-link"
          href="/cart"
        >
          Cart
        </a>
      </div>
    </div>
  </div>
</nav>
`;
